examples/DrawPrimitives/minimalapp.cpp

```diff
@@ -3,76 +3,107 @@
+    m_engine.reset( nullptr );
+}
+
+void MinimalApp::initialize() {
+    QSurfaceFormat format;
+    format.setVersion( 4, 4 );
+    format.setProfile( QSurfaceFormat::CoreProfile );
+    format.setDepthBufferSize( 24 );
+    format.setStencilBufferSize( 8 );
+    format.setSamples( 16 );
+    format.setSwapBehavior( QSurfaceFormat::DoubleBuffer );
+    format.setSwapInterval( 0 );
+    QSurfaceFormat::setDefaultFormat( format );
 
     // Initialize Engine.
-    m_engine.reset( Ra::Engine::RadiumEngine::createInstance() );
+    m_engine.reset( Engine::RadiumEngine::createInstance() );
     m_engine->initialize();
 
-    ///\todo update when a basic viewer is implemented ... (to call setupKeyMappingCallbacks)
-    Ra::Gui::KeyMappingManager::createInstance();
-    Ra::Gui::KeyMappingManager::getInstance()->addListener(
-        Ra::Gui::TrackballCameraManipulator::configureKeyMapping );
-    Ra::Gui::KeyMappingManager::getInstance()->addListener( Ra::Gui::Viewer::configureKeyMapping );
-
     // Initialize taskqueue.
-    m_task_queue.reset( new Ra::Core::TaskQueue( std::thread::hardware_concurrency() - 1 ) );
-    // Initialize viewer.
-    m_viewer.reset( new Ra::Gui::Viewer );
+    m_taskQueue.reset( new Core::TaskQueue( std::thread::hardware_concurrency() - 1 ) );
 
+    // Initialize viewer.
+    m_viewer.reset( new Viewer );
     CORE_ASSERT( m_viewer != nullptr, "GUI was not initialized" );
+
+    m_viewer->setupKeyMappingCallbacks();
+    auto keyMappingManager = KeyMappingManager::getInstance();
+    keyMappingManager->addListener(
+        RotateAroundCameraManipulator::KeyMapping::configureKeyMapping );
+
     connect( m_viewer.get(),
-             &Ra::Gui::Viewer::requestEngineOpenGLInitialization,
+             &Viewer::requestEngineOpenGLInitialization,
              this,
-             &MinimalApp::onGLInitialized );
-    m_viewer->setupKeyMappingCallbacks();
+             &MinimalApp::onRequestEngineOpenGLInitialization );
+    connect( m_viewer.get(), &Viewer::glInitialized, this, &MinimalApp::onGLInitialized );
+    connect( this, &QGuiApplication::lastWindowClosed, m_viewer.get(), &WindowQt::cleanupGL );
+
+    CORE_ASSERT( m_viewer->getContext() != nullptr, "OpenGL context was not created" );
+    CORE_ASSERT( m_viewer->getContext()->isValid(), "OpenGL was not initialized" );
 
     // Initialize timer for the spinning cube.
-    m_frame_timer = new QTimer( this );
-    m_frame_timer->setInterval( 1000 / m_target_fps );
-}
+    m_frameTimer = new QTimer( this );
+    m_frameTimer->setInterval( 1000 / m_targetFps );
 
-MinimalApp::~MinimalApp() {
-    // need to clean up everithing before engine is cleaned up.
-    m_task_queue.reset( nullptr );
-    m_viewer.reset( nullptr );
-    m_engine->cleanup();
-    m_engine.reset( nullptr );
+    // Create a window container for the viewer.
+    // using viewer directly as main windows fail to have valid glbinding context
+    // on Debian bullseye, Qt version 6.2.4
```

Comment:
I'm not sure this is only related to Debian bullseye but rather to the Qt version (and its implementation ?). See SimpleWindow::SimpleWindow, added by the PR #608, that do the same.

Reply:
Well, it works on debian + qt 5, so it's related at least to qt version, and if current release-candidate works on macos, it may be related to the qt os specific implementation.